### PR TITLE
Send log messages to the database

### PR DESCRIPTION
**Pull request by Christian-B**
send all log messages to the provenance databse.

Must be done at the same time as:
https://github.com/SpiNNakerManchester/SpiNNUtils/pull/174

This includes safety code if the database is locked.

Tested by https://github.com/SpiNNakerManchester/IntegrationTests/pull/137

### CODE REVIEW

File: spinn_front_end_common/interface/provenance/db.sql [hunk 1]
```diff
@@ -218,3 +218,31 @@ CREATE TABLE IF NOT EXISTS boards_provenance(
...
+VALUES
+    (50, "CRITICAL"),
+    (40, "ERROR"),
+    (30, "WARNING"),
+    (20, "INFO"),
+    (10, "DEBUG");
+
+CREATE VIEW IF NOT EXISTS p_log_view AS
+    SELECT
+		name,
```
Comment (dkfellows) [inline spinn_front_end_common/interface/provenance/db.sql:244]: Use tabs or don't use tabs, but don't mix.

File: spinn_front_end_common/interface/provenance/db.sql [hunk 2]
```diff
@@ -218,3 +218,31 @@ CREATE TABLE IF NOT EXISTS boards_provenance(
     ip_addres STRING NOT NULL,
     ethernet_x INTEGER NOT NULL,
     ethernet_y INTEGER NOT NULL);
+
+---------------------------------------------------------------------
+-- A table to store log.info
+CREATE TABLE IF NOT EXISTS p_log_provenance(
```
Comment (dkfellows) [inline spinn_front_end_common/interface/provenance/db.sql:224]: Should this table include the timestamp as its own column?

Comment (Christian-B) [reply spinn_front_end_common/interface/provenance/db.sql:224]: good question I dont know

File: spinn_front_end_common/interface/provenance/db.sql (hunk 1 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/provenance/db.sql:244]: fixed

**Comment by Christian-B** [comment]
> Is what is being logged controlled by the general logging level? I ask because logging stuff at `DEBUG` level is likely to produce a lot of write activity (and may have quite a profound effect on performance).

Currently yes it does . because I had not considered that.

An issue is that in FormatAdapter.log has self.isEnabledFor(level) even for level 10 (debug)

**Comment by Christian-B** [comment]
If https://github.com/SpiNNakerManchester/SpiNNUtils/pull/177 is approved it will be easy to only write to database if the message is of interest.

File: spinn_front_end_common/interface/provenance/db.sql (hunk 2 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/provenance/db.sql:224]: Done

**Review by rowleya — APPROVED** [review]

**Pull request merged**